parser/ParseSample.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * This file copyright (c) 2011-2015, Quickstep Technologies LLC.
+ * All rights reserved.
+ * See file CREDITS.txt for details.
+ **/
+
+#ifndef QUICKSTEP_PARSER_PARSE_SAMPLE_HPP_
+#define QUICKSTEP_PARSER_PARSE_SAMPLE_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parser/ParseLiteralValue.hpp"
+#include "parser/ParseTreeNode.hpp"
+#include "utility/Macros.hpp"
+
+namespace quickstep {
+
+/** \addtogroup Parser
+ *  @{
+ */
+
+/**
+ * @brief A parsed representation of BLOCK SAMPLE.
+ */
+class ParseSample : public ParseTreeNode {
+ public:
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number The line number of " SAMPLE" in the SQL statement.
+   * @param column_number The column number of " SAMPLE" in the SQL statement.
+   * @param limit_expression The  SAMPLE value expression.
+   */
+  ParseSample(int line_number, int column_number, bool is_block_sample,NumericParseLiteralValue *percentage)
```

Comment:
Since there are four arguments, better format them as aligned lines, i.e.
```
ParseSample(const int line_number,
            const int column_number,
            const bool is_block_sample,
            NumericParseLiteralValue *percentage)
```
(also add `const` as above shows)